src/cpu/x64/injectors/jit_uni_binary_injector.cpp

```diff
@@ -2647,22 +2647,57 @@ template <typename T>
 typename std::enable_if<!(std::is_same<T, Xbyak::Zmm>::value
                           || std::is_same<T, Xbyak::Address>::value)>::type
 jit_uni_binary_injector_t<isa, Vmm>::execute_prelu_binary(const Vmm &dst,
-                                                        const Vmm &lhs, const T &rhs) const {
-    // in sse4 vmm_aux0 as mask it's index must be 0
-    Vmm vmm_aux0 = Vmm(rhs_arg_static_params_.rhs_prelu_helper_vmm_idx);
-
-    if (dst == vmm_aux0) {
-        vmm_aux0 = Vmm(14);
-        if (isa == sse41)
-            assert(!"conflict mask register");
-    }
+    const Vmm &lhs, const T &rhs) const {
+    if (is_superset(isa, avx)) {
+        host_->uni_vmulps(rhs, rhs, lhs);
+        host_->uni_vblendvps(dst, lhs, rhs, lhs);
+    } else {
+        using dnnl::impl::utils::one_of;
+        // in sse4 vmm_aux0 as mask it's index must be 0
+        Vmm vmm_aux0 = Vmm(rhs_arg_static_params_.rhs_prelu_helper_vmm_idx);
+
+        if (one_of(vmm_aux0, dst, lhs, rhs)) {
+            //let's find a vacant XMM register
+            int occupied_idices[] = {dst.getIdx(), lhs.getIdx(), rhs.isMEM() ? -1 : rhs.getIdx()};
```

Comment:
rhs should not be memory in my understanding for this template function.
 **std::enable_if<!(std::is_same<T, Xbyak::Zmm>::value
                          || std::is_same<T, Xbyak::Address**

Reply:
Done